File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());
        int count = 0;
        sb.append("<");

        Queue<Integer> queue = new LinkedList<>();

        for (int i = 1; i <= N; i++) {
            queue.offer(i);
        }

        while (!queue.isEmpty()) {
            count += 1;
            int first = queue.poll();

            if (count % K == 0) {
                sb.append(first).append(", ");
            } else {
                queue.offer(first);
            }
        }

            sb.delete(sb.length() - 2, sb.length());


        sb.append(">");
        System.out.println(sb);
    }

}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys
from collections import deque

N, K = map(int, sys.stdin.readline().split())

queue = deque()
answers = "<"
count = 0

for i in range(1, N+1):
    queue.append(i)


while len(queue) != 0:
    count += 1
    first = queue.popleft()
    if count % K == 0:
        answers += str(first)
        if len(queue) != 0:
            answers += ", "
    else:
        queue.append(first)
    

print(answers + ">")